sycl/test-e2e/NonUniformGroups/ballot_group_algorithms.cpp

```diff
@@ -102,6 +104,24 @@ int main() {
           uint32_t IncScanResult =
               sycl::inclusive_scan_over_group(BallotGroup, 1, sycl::plus<>());
           IncScanAcc[WI] = (IncScanResult == LID + 1);
+
+          uint32_t ShiftLeftResult =
+              sycl::shift_group_left(BallotGroup, LID, 2);
```

Comment:
Same question for all the tests: is there a reason to shift by 2 instead of by 1?

I'm a little concerned that this might hide a bug.  If we shift these groups by 1, it should be equivalent to shifting the entire sub-group by 2.  Having a shift of 2 here might be misinterpreted as using the original sub-group numbering.

Reply:
The reason I picked 2 was to avoid testing the default behavior of shift. I would argue it is more representative than just moving by one item, but given the nature of the test I think both are very interesting. Would you prefer we go with 1?

Reply:
Oh, I see.  You've convinced me that checking 2 makes sense.

I think we need to do a lot more testing to cover the corner-cases, but we can do that in the CTS.  For example, I think we need to check that things don't blow up if the two "halves" of a ballot group try and shift by different amounts.